clients/src/main/java/org/apache/kafka/common/security/JaasUtils.java

```diff
@@ -58,6 +58,11 @@ public static boolean isZkSecurityEnabled(String loginConfigFile) {
                 throw new KafkaException("Exception while determining if ZooKeeper is secure");
             }
         }
+        /*
+         * Tests fail if we don't reset the login configuration. It is unclear
+         * what is actually triggering this bug.
+         */
+        Configuration.setConfiguration(null);
```

Comment:
As we were discussing with @junrao, I think we need to either find a way not to need this or to at least do it in a safe way within our app (ie reload it atomically).